### PR TITLE
Remove "Usage: await x();" help text in QuickInfo and SigHelp

### LINKED ISSUE #39779: Minor, the async usage "example" in some tooltips is wrong
**Version Used**: 16.3.5

**Screenshot of the Issue**:

If you mouse over the `async` keyword when it's applied to a lambda, you get this:
image

That usage example otherwise shows `Usage: await functionname(...);`, and while I generally find all of these usage examples useless, if they're there they might as well be correct. `await()` is not even valid syntax.

### PR DESCRIPTION
**Pull request by dpoeschl**
Fixes #39779

### CODE REVIEW

File: src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs [hunk 1]
```diff
@@ -62,14 +62,12 @@ protected static SymbolDisplayPart NewLine()
                 parameter.ToMinimalDisplayParts(semanticModel, position));
         }
 
+        /// <summary>
+        /// We no longer show awaitable usage text in SignatureHelp, but IntelliCode expects this
+        /// method to exist.
```
Comment (jasonmalinowski) [inline src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs]: Can we mark this as [Obsolete] now?

Comment (jasonmalinowski) [reply src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs]: (Look at other Obsolete attributes we have in the repo, we have a pattern for "this is a backcompat shim for <other thing>", which makes it easy to grep for things we can clean up.)

File: src/EditorFeatures/VisualBasicTest/QuickInfo/SemanticQuickInfoSourceTests.vb [hunk 2]
```diff
@@ -1419,10 +1419,8 @@ End Class
 
             Dim description = <File>&lt;<%= VBFeaturesResources.Awaitable %>&gt; Function C.goo() As Task</File>.ConvertTestSourceTag()
 
-            Dim doc = StringFromLines("", WorkspacesResources.Usage_colon, $"  {SyntaxFacts.GetText(SyntaxKind.AwaitKeyword)} goo()")
-
             Await TestFromXmlAsync(markup,
-                 MainDescription(description), Usage(doc))
+                 MainDescription(description), Usage(String.Empty))
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/QuickInfo/SemanticQuickInfoSourceTests.vb]: 💭 Can we not simplify this to omit the call to `Usage`?

File: src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs [hunk 3]
```diff
@@ -4370,26 +4370,6 @@ class C<T>
                 MainDescription(@"class C<T>"));
         }
 
-        [Fact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
-        public async Task TestAwaitableMethod()
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/QuickInfo/SemanticQuickInfoSourceTests.cs]: 💭 Maybe keep this method but remove `documentation` from it?

**Pull request merged**